shared/js/background/safari-events.es6.js

```diff
@@ -0,0 +1,295 @@
+const ATB = require('./atb.es6')
+const tabManager = require('./tab-manager.es6')
+const utils = require('./utils.es6')
+const Companies = require('./companies.es6')
+const settings = require('./settings.es6')
+const abpLists = require('./abp-lists.es6')
+
+// add these to contentWindow so the popup can access them
+safari.extension.globalPage.contentWindow.tabManager = tabManager
+safari.extension.globalPage.contentWindow.Companies = Companies
```

Comment:
Is it not possible to do messages between the popup and the background processes? I know they run in the same context, but it would make things loads cleaner, and would probably help us standardize between the Chrome/FF and Safari a bit more.

Reply:
Nope we can't message the background. That's one of the big complications with Safari.

Reply:
Potentially stupid idea, what if we faked a messaging interface? E.g. off the top of my head:

```js
// in UI code
safari.extension.globalPage.contentWindow.sendMessage({ getSetting: 'foo' })
   .then(() => { bar() })

// in background code
safari.extension.globalPage.contentWindow.sendMessage = (message) => {
    if (message.getSetting) {
        return Promise.resolve(valueFromSettings)
    }
}
```

The benefit will be that at least Safari and Chrome/FF will communicate in a similar way.

Reply:
It's a good idea. Let's save that for a follow up. I'd want to spend some time cleaning up `fetch` to work with that too.